stored_requests/config/config_test.go

```diff
@@ -41,12 +41,42 @@ func isMemoryCacheType(cache stored_requests.CacheJSON) bool {
 }
 
 func TestNewEmptyFetcher(t *testing.T) {
-	fetcher := newFetcher(&config.StoredRequests{}, nil, nil)
-	if fetcher == nil {
-		t.Errorf("The fetcher should be non-nil, even with an empty config.")
+
+	type testCase struct {
+		config  *config.StoredRequests
+		message string
+	}
+	testCases := []testCase{
+		{
+			config:  &config.StoredRequests{},
+			message: "If the config is empty, and EmptyFetcher should be returned",
+		},
+		{
+			config: &config.StoredRequests{
+				Postgres: config.PostgresConfig{
+					CacheInitialization: config.PostgresCacheInitializer{
+						Query: "test query",
+					},
+					PollUpdates: config.PostgresUpdatePolling{
+						Query: "test poll query",
+					},
+					FetcherQueries: config.PostgresFetcherQueries{
+						QueryTemplate: "",
+					},
+				},
+			},
+			message: "If Postgres fetcher query is not defined, but Postgres Cache init query and Postgres update polling query are defined EmptyFetcher should be returned",
```

Comment:
Can we please also add test cases where `PostgresFetcherQueries` is supplied in the config but not `PostgresUpdatePolling` and `PostgresCacheInitializer`? And also one where all of them are supplied?

Reply:
Yes, added!